run_unittests.py

```diff
@@ -100,15 +67,34 @@ def setup_backend():
     os.environ['MESON_UNIT_TEST_BACKEND'] = be
     sys.argv = filtered
 
+def import_test_cases(suite: unittest.TestSuite) -> list:
```

Comment:
I think this is a list of test cases, right? Although I wonder if it would make more sense because we're runnign this recursively to just return a `set[TestCase]` and have the caller sort, or to have a private implementation that does all set work, and a public function that just calls `sorted(_private(...))`.

```suggestion
def import_test_cases(suite: unittest.TestSuite) -> T.List[unittest.TestCase]:
```

Reply:
ok, I went with a set now. I just wanted to not change any types compared to before.